### PR TITLE
Load FundingForm properly in reconfig

### LINKED ISSUE #827: Reconfig modal not loading correct distribution limit
## Summary

I have a project with 1eth distribution limit, but if I open the reconfig modal it shows 'Zero' limit 

## Environment

- **Device/OS**: [e.g. iOS]
- **Browser:** [e.g. chrome, safari]
- **Hardware wallet**: [e.g. Ledger w/ Metamask]

### LINKED ISSUE #828: Considerations for zero funding cycle duration V2
## Summary

Screen Shot 2022-04-18 at 12 12 06 pm

- Upcoming funding cycle tab should be hidden when duration is zero
- Reconfigure button should be changed to just 'Reconfigure'

## Environment

- **Device/OS**: [e.g. iOS]
- **Browser:** [e.g. chrome, safari]
- **Hardware wallet**: [e.g. Ledger w/ Metamask]

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

- After redoing the V2ProjectContext, the fundingForm on the reconfig was no longer loading upcoming stuff. Fixed this (Closes #827 )
- Hide upcoming tab and reword 'Reconfigure upcoming' -> 'Reconfigure' when duration is zero:

Screen Shot 2022-04-18 at 5 55 27 pm

(Closes #828 )

- Minor spelling mistake fixes brought to my attention by translators 

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

File: src/components/v2/V2Create/ProjectPreview.tsx [hunk 1]
```diff
@@ -55,7 +56,10 @@ export default function ProjectPreview({
     fundingCycleMetadata,
 
     distributionLimit: fundAccessConstraint?.distributionLimit,
-    distributionLimitCurrency: fundAccessConstraint?.distributionLimitCurrency,
+    distributionLimitCurrency:
```
Comment (aeolianeth) [inline src/components/v2/V2Create/ProjectPreview.tsx:59]: **question** why this change?

File: src/components/v2/V2Project/V2FundingCycleSection/index.tsx [hunk 2]
```diff
@@ -31,6 +36,13 @@ export default function V2FundingCycleSection({
   } = useContext(ThemeContext)
   const { fundingCycle, isPreviewMode } = useContext(V2ProjectContext)
 
+  const canReconfigure = useHasPermission(V2OperatorPermission.RECONFIGURE)
+  const showReconfigureButton = canReconfigure && !isPreviewMode
+
+  if (!fundingCycle) {
```
Comment (aeolianeth) [inline src/components/v2/V2Project/V2FundingCycleSection/index.tsx:42]: **issue**: can we add a loading flag for `fundingCycleLoading` in V2PorjectContext?

File: src/components/v2/V2Create/forms/FundingForm/index.tsx [hunk 3]
```diff
@@ -78,31 +92,95 @@ export default function FundingForm({ onFinish }: { onFinish: VoidFunction }) {
...
+    domain: effectiveFundingCycle?.configuration?.toString(),
+  })
+
   const ETHPaymentTerminalFee = useETHPaymentTerminalFee()
   const feeFormatted = ETHPaymentTerminalFee
     ? formatFee(ETHPaymentTerminalFee)
     : undefined
 
+  // Load queued FC data (if it exists) into redux
+  useEffect(() => {
```
Comment (aeolianeth) [inline src/components/v2/V2Create/forms/FundingForm/index.tsx:141]: @johnnyd-eth I was actually thinking that all this default-redux-setting stuff would live in either:
- src/components/v2/V2Project/V2ProjectReconfigureModal/index.tsx
- src/components/v2/V2Project/V2ProjectReconfigureModal/V2ReconfigureModalTrigger.tsx

the reason being that queuedFC stuff isn't relevant to the create flow, so we don't need all this extra code in this FundingForm component.

WDYT

**Review by aeolianeth — CHANGES_REQUESTED** [review]

File: src/components/v2/V2Create/forms/FundingForm/index.tsx (hunk 3 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Create/forms/FundingForm/index.tsx:141]: Yeah true ok agreed

File: src/components/v2/V2Create/ProjectPreview.tsx (hunk 1 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Create/ProjectPreview.tsx:59]: Yeah this was lazy. Was seeing distributionLimitCurrency was coming in as 0 so it was fucking up some v2 stuff. So I changed the default redux state to use distributionLimitCurrency == V2_CURRENCY_ETH

File: src/components/v2/V2Project/V2FundingCycleSection/index.tsx (hunk 2 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Project/V2FundingCycleSection/index.tsx:42]: Yep

File: src/redux/slices/editingV2Project.ts [hunk 4]
```diff
@@ -96,7 +109,7 @@ export const defaultProjectState: V2ProjectState = {
   projectMetadata: { ...defaultProjectMetadataState },
   fundingCycleData: { ...defaultFundingCycleData },
   fundingCycleMetadata: { ...defaultFundingCycleMetadata },
-  fundAccessConstraints: [],
+  fundAccessConstraints: [defaultFundAccessConstraint],
```
Comment (aeolianeth) [inline src/redux/slices/editingV2Project.ts:112]: are you sure we need this? deploys seem fine without it.

File: src/components/v2/V2Project/V2FundingCycleSection/index.tsx [hunk 5]
```diff
@@ -59,22 +79,23 @@ export default function V2FundingCycleSection({
...
       label: tabText({ text: t`Current` }),
       content: <CurrentFundingCycle expandCard={expandCard} />,
     },
-    !isPreviewMode && {
-      key: 'upcoming',
-      label: tabText({ text: t`Upcoming` }),
-      content: <UpcomingFundingCycle expandCard={expandCard} />,
-    },
+    !isPreviewMode &&
+      fundingCycle &&
```
Comment (aeolianeth) [inline src/components/v2/V2Project/V2FundingCycleSection/index.tsx:91]: `fundingCycle` is already guarded against on line 50, i thinkwe can remove this check here

File: src/components/v2/V2Project/V2ProjectReconfigureModal/V2ReconfigureModalTrigger.tsx [hunk 6]
```diff
@@ -29,6 +41,68 @@ export default function V2ReconfigureFundingModalTrigger({
     setReconfigureModalVisible(true)
   }
 
+  // Load queued FC data of project
+  const { data: queuedFundingCycleResponse } = useProjectQueuedFundingCycle({
+    projectId,
+  })
+
+  const [queuedFundingCycle] = queuedFundingCycleResponse || []
```
Comment (aeolianeth) [inline src/components/v2/V2Project/V2ProjectReconfigureModal/V2ReconfigureModalTrigger.tsx:49]: does `??` work here?

File: src/contexts/v2/projectContext.ts [hunk 7]
```diff
@@ -9,6 +9,7 @@ type V2ProjectLoadingStates = {
   ETHBalanceLoading: boolean
   balanceInDistributionLimitCurrencyLoading: boolean
   distributionLimitLoading: boolean
+  fundingCycleLoading: boolean
```
Comment (aeolianeth) [inline src/contexts/v2/projectContext.ts:12]: fresh

File: src/components/v2/V2Create/forms/FundingForm/index.tsx [hunk 8]
```diff
@@ -78,31 +78,33 @@ export default function FundingForm({ onFinish }: { onFinish: VoidFunction }) {
...
 
+  // Load redux state (will be empty in create flow)
+  const { fundAccessConstraints, fundingCycleData, payoutGroupedSplits } =
+    useAppSelector(state => state.editingV2Project)
+  const fundAccessConstraint =
+    getDefaultFundAccessConstraint<SerializedV2FundAccessConstraint>(
+      fundAccessConstraints,
+    )
+
+  // Loads redux state into form
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/components/v2/V2Create/forms/FundingForm/index.tsx:97]: I think it's pretty accurate?

File: src/components/v2/V2Project/V2FundingCycleSection/index.tsx [hunk 9]
```diff
@@ -29,7 +34,22 @@ export default function V2FundingCycleSection({
...
+  } = useContext(V2ProjectContext)
+
+  const canReconfigure = useHasPermission(V2OperatorPermission.RECONFIGURE)
+  const showReconfigureButton = canReconfigure && !isPreviewMode
+
+  if (fundingCycleLoading) {
+    return <Loading />
+  }
+
+  if (!fundingCycle) {
```
[the remark this answers is not in the document]
Comment (johnnyd-eth) [reply src/components/v2/V2Project/V2FundingCycleSection/index.tsx:50]: Yeah good idea

File: src/components/v2/V2Project/V2FundingCycleSection/index.tsx (hunk 5 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Project/V2FundingCycleSection/index.tsx:91]: Yup good catch

**Comment by aeolianeth** [comment]
MEERRRGGGGING

**Review by aeolianeth — APPROVED** [review]
